library/old_protocol/deserialize.cpp

```diff
@@ -75,18 +75,13 @@ int unpack(dnet_net_state *st, const data_pointer &data, T &value, size_t &lengt
 
 int deserialize_lookup_request(dnet_net_state *st, const dnet_cmd &cmd,
                                std::unique_ptr<n2_request> &out_deserialized) {
-	out_deserialized.reset(new(std::nothrow) lookup_request(cmd));
-	if (!out_deserialized)
-		return -ENOMEM;
-
+	out_deserialized.reset(new lookup_request(cmd));
```

Comment:
This also now could return value.

Reply:
Deserialization usually returns error code. May be similar functions will be better for using?

Reply:
The logic will be reworked in next commits: we'll have deserializers only for bodies, so they all could return error (some bad body format, e.g. length error). In other words, deserializers became more similar =)

Reply:
(fixed in next commits)